crates/red_knot_python_semantic/src/semantic_index/symbol.rs

```diff
@@ -475,6 +475,13 @@ impl NodeWithScopeKind {
             _ => panic!("expected type alias"),
         }
     }
+
+    pub const fn as_function(&self) -> Option<&ast::StmtFunctionDef> {
+        match self {
+            Self::Function(function) => Some(function.node()),
+            _ => None,
+        }
+    }
```

Comment:
- Elsewhere in red-knot we use the `into_` prefix for functions like this -- should we do the same here? E.g. https://github.com/astral-sh/ruff/blob/5bc9d6d3aa694ab13f38dd5cf91b713fd3844380/crates/red_knot_python_semantic/src/types.rs#L566-L571
- You can simplify the `expect_function` method a few lines above now that you've added this method:

    ```diff
    diff --git a/crates/red_knot_python_semantic/src/semantic_index/symbol.rs b/crates/red_knot_python_semantic/src/semantic_index/symbol.rs
	index 3e1ec2806..5ed9746e0 100644
	--- a/crates/red_knot_python_semantic/src/semantic_index/symbol.rs
	+++ b/crates/red_knot_python_semantic/src/semantic_index/symbol.rs
	@@ -463,10 +463,7 @@ impl NodeWithScopeKind {
	     }
	 
	     pub fn expect_function(&self) -> &ast::StmtFunctionDef {
	-        match self {
	-            Self::Function(function) => function.node(),
	-            _ => panic!("expected function"),
	-        }
	+        self.as_function().expect("Expected a function")
	     }
    ```

Reply:
It's called `as` because it doesn't consume `self`, consistent with the terminology we use elsewhere.